viewing_party/party.py

```diff
@@ -1,23 +1,184 @@
+  genres_list = []
+  genres_dict = {}
+  count = 0
+  for movie_list in user_data["watched"]:
+    if len(movie_list) != 0:
+      genre_title = movie_list["genre"]   
+      if genre_title in genres_list:
+        count += 1
+        genres_dict[count] = count
+      else:
+        genres_list.append(genre_title)
+        genres_dict[genre_title] = 1
+      
+        return genres_list[0]
+        
+    return None
+      
+  
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+  list_of_user_movies = []
+  list_of_friend_movies = []
+  user_has_watched_freinds_have_not_list = []
+  for movie_list in user_data["watched"]:
+    list_of_user_movies.append(movie_list)
+  
+  for friend_movie_list in user_data["friends"]:
+    for watched_friend_list in friend_movie_list["watched"]:
+      list_of_friend_movies.append(watched_friend_list)
+  for title in list_of_user_movies:
+    if title not in list_of_friend_movies:
+      user_has_watched_freinds_have_not_list.append(title)
+  return user_has_watched_freinds_have_not_list
+
 
+def get_friends_unique_watched(user_data):
+  list_of_user_movies = []
+  list_of_friend_movies = []
+  movies_one_of_users_freinds_have_watched_but_user_has_not = []
+  
+  for movie_list in user_data["watched"]:
+    list_of_user_movies.append(movie_list)
+  
+  for friend_movie_list in user_data["friends"]:
+    for watched_friend_list in friend_movie_list["watched"]:
+      list_of_friend_movies.append(watched_friend_list)
+  # movies_one_of_users_freinds_have_watched_but_user_has_not = []
+  
+  for movie in (list_of_friend_movies):
+    if movie not in list_of_user_movies and len(movie) > 0:
+      if movie not in movies_one_of_users_freinds_have_watched_but_user_has_not:
+        movies_one_of_users_freinds_have_watched_but_user_has_not.append(movie)
+      
+  return movies_one_of_users_freinds_have_watched_but_user_has_not
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+  list_of_user_movies = []
+  list_of_friend_movies = []
+  rec_list = []
+  
+  for movie_list in user_data["watched"]:
+    list_of_user_movies.append(movie_list)
+  
+  for friend_movie_list in user_data["friends"]:
+    for watched_friend_list in friend_movie_list["watched"]:
+      list_of_friend_movies.append(watched_friend_list)
+
+  for movie in (list_of_friend_movies):
+    if movie not in list_of_user_movies and len(movie) > 0:
+      if movie not in rec_list:
+        if movie["host"] in user_data["subscriptions"]:
+          rec_list.append(movie)
+        
+  return rec_list
+    
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):

```

Comment:
As with `get_available_recs`, this implementation is very similar to `get_friends_unique_watched`. So we can look for a way to reuse that function to simplify this code.